src/react/components/pages/train/trainPage.tsx

```diff
@@ -362,10 +366,9 @@ export default class TrainPage extends React.Component<ITrainPageProps, ITrainPa
 
             return trainStatusRes;
         } catch (error) {
-            const isOnPrem = isElectron && this.props.project.sourceConnection.providerType === "localFileSystemProxy";
             this.setState({
                 showTrainingFailedWarning: true,
-                trainingFailedMessage: isOnPrem ? interpolate(strings.train.errors.electron.cantAccessFiles, { folderUri: this.state.inputtedLabelFolderURL }) :
+                trainingFailedMessage: this.isOnPrem ? interpolate(strings.train.errors.electron.cantAccessFiles, { folderUri: this.state.inputtedLabelFolderURL }) :
```

Comment:
Nested ternary operations is not good coding practice. Should just use a switch or an if else